lib/vsc/install/shared_setup.py

```diff
@@ -0,0 +1,548 @@
+
+    if isinstance(v, list):
+        return ",".join(v)
+
+
+def parse_target(target):
+    """Add some fields"""
+    new_target = {}
+    new_target.update(SHARED_TARGET)
+    for k, v in target.items():
+        if k in ('author', 'maintainer'):
+            if not isinstance(v, list):
+                log.error("%s of config %s needs to be a list (not tuple or string)" % (k, target['name']))
+                sys.exit(1)
+            new_target[k] = ";".join([x[0] for x in v])
+            new_target["%s_email" % k] = ";".join([x[1] for x in v])
+        else:
+            if isinstance(v, dict):
+                # eg command_class
+                if not k in new_target:
+                    new_target[k] = type(v)()
+                new_target[k].update(v)
+            else:
+                new_target[k] = type(v)()
+                new_target[k] += v
+
+    log.debug("New target = %s" % (new_target))
+    return new_target
+
+
+def build_setup_cfg_for_bdist_rpm(target):
+    """Generates a setup.cfg on a per-target basis.
+
+    Stores the 'install-requires' in the [bdist_rpm] section
+
+    @type target: dict
+
+    @param target: specifies the options to be passed to setup()
+    """
+
+    try:
+        setup_cfg = open('setup.cfg', 'w')  # and truncate
+    except (IOError, OSError), err:
+        print "Cannot create setup.cfg for target %s: %s" % (target['name'], err)
+        sys.exit(1)
+
+    s = ["[bdist_rpm]"]
+    if 'install_requires' in target:
+        s += ["requires = %s" % (sanitize(target['install_requires']))]
+
+    if 'provides' in target:
+        s += ["provides = %s" % (sanitize((target['provides'])))]
+        target.pop('provides')
+
+    setup_cfg.write("\n".join(s) + "\n")
+    setup_cfg.close()
+
+
+def action_target(target, setupfn=setup, extra_sdist=[], urltemplate=None):
+    # EXTRA_SDIST_FILES.extend(extra_sdist)
+
+    do_cleanup = True
+    try:
+        # very primitive check for install --skip-build
+        # in that case, we don't mind "leftover build";
+        # it's probably intentional
+        install_ind = sys.argv.index('install')
+        build_skip = sys.argv.index('--skip-build')
+        if build_skip > install_ind:
+            do_cleanup = False
+    except ValueError:
+        pass
+
+    if do_cleanup:
+        cleanup()
+
+    if urltemplate:
+        target['url'] = urltemplate % target
+        if 'github' in urltemplate:
+            target['download_url'] = "%s/tarball/master" % target['url']
+
+    build_setup_cfg_for_bdist_rpm(target)
+    x = parse_target(target)
+
+    setupfn(**x)
+
+if __name__ == '__main__':
+    """
+    This main is the setup.py for vsc-install
+    """
+    PACKAGE = {
+        'name': 'vsc-install',
+        'version': '1.0.0',
+        'author': [sdw, ag, jt],
+        'maintainer': [sdw, ag, jt],
+        'packages': ['vsc', 'vsc.install'],
+        'zip_safe': True,
+    }
+
+    action_target(PACKAGE, urltemplate=URL_GH_HPCUGENT)
```

Comment:
might as well make `URL_GH_HPCUGENT` the default value for `urltemplate` (see https://github.com/hpcugent/vsc-base/pull/203)